plugin-server/src/cdp/cdp-consumers.ts

```diff
@@ -161,8 +183,13 @@ abstract class CdpConsumerBase {
                         const logs = result.logs
                         result.logs = []
 
-                        counterFunctionInvocation.inc({
-                            outcome: result.error ? 'failed' : 'succeeded',
+                        this.logAppMetrics({
+                            team_id: result.teamId,
+                            app_source_id: result.hogFunctionId,
+                            instance_id: result.id,
```

Comment:
If this is a per-invocation ID, like a UUID, it will prevent a lot of (almost all?) MergeTree aggregation.

Is querying by invocation a product requirement? If not, probably better to use `''` if Hog really has no equivalent here...

Although I kind of assumed Hog apps could be (or eventually would be) reused across teams and/or instantiated multiple times within the same time, and so `app_source_id` would be some stable app_id and `instance_id` would reflect the individual instantiation of the app?

Reply:
(Requesting changes to make sure this is seen, feel free to override if I'm missing something of course.)

Reply:
Yeah I was wondering about that. I don't have any use case for it. For logs `instance_id` is used as the invocation instance which makes sense but for app metrics its basically useless. I'll remove it.

Reply:
If there's no stable ID for the app (and not the instance) then another idea is:

`app_source_id=hogfunction id`
`instance_id=hash of function source code at the time`

This lets you track metric changes between code changes, and preferably we keep a history of all versions somewhere (?) so the code for the point in time can be looked up.

Reply:
There's no such concept as an app. Every function is just source code but they are derived from a starter template. Once it is created there is no link to the original source (at least no automatic link).

Hash could be a good idea 🤔 Gives us a way of conceptually having "versions". That would need some more thinking though so I'll leave that for a follow up

Reply:
Yeah, definitely a followup, I'll just say a stable hash of the source (and way to lookup source from a hash) was something I used multiple times per week running Workers.